### PR TITLE
actions: fix duplicated builds

### LINKED ISSUE #103: Github Actions: avoid dual build problem
At the moment we are triggering a dual build for internal pull requests (a PR originating from this repo, to master branch for example).

After a quick search, it's possible to avoid this. examples:
- https://github.com/decidim/decidim/pull/5883
- https://github.com/microsoft/vscode/pull/83158

I recommend to limit the `push` to `master` branch and let `pull_request` trigger on any branches.

### PR DESCRIPTION
**Pull request by Wenzel**
fix #103

### CODE REVIEW

**Comment by rageagainsthepc** [comment]
So what happens when a branch is pushed after a PR has been created? Afaik this is not automatically retested, right?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 name: Build
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - "*"
```
Comment (rageagainsthepc) [inline .github/workflows/build.yml:9]: nitpick: You don't need this afaik.

**Comment by Wenzel** [comment]
> So what happens when a branch is pushed after a PR has been created?

As far as I understand, since the PR is opened, any push on the branch will trigger the `pull_request` event.
The goal is to avoid the `push` event to be triggered also.
https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request

File: .github/workflows/build.yml (hunk 1 above)
Comment (Wenzel) [reply .github/workflows/build.yml:9]: so an empty key `pull_request` is avalid you think ?
I can try...

**Comment by Wenzel** [comment]
@rageagainsthepc also, see just now,
we have pushed a new commit and the pull_request event has been triggered, even though the PR was already opened.

File: .github/workflows/build.yml (hunk 1 above)
Comment (Wenzel) [reply .github/workflows/build.yml]: uh, you were right

**Comment by Wenzel** [comment]
@rageagainsthepc  ?

**Comment by rageagainsthepc** [comment]
I could swear that at some point feature branches were not retested in PRs if you only had the on pull request action. Glad to see that there is a solution that avoids building twice though. ;) This should also fix clippy annotations being posted twice, right?

**Pull request merged**